testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jaxrs/integration/cdi/CDIBean.java

```diff
@@ -4,9 +4,12 @@
  */
 package org.jboss.as.test.integration.jaxrs.integration.cdi;
 
+import jakarta.enterprise.context.ApplicationScoped;
+
 /**
  * @author Stuart Douglas
  */
+@ApplicationScoped
```

Comment:
This annotation is required to make the `CDIBean` a real CDI bean if the `bean-discovery-mode` is set to `annotated`.